glib-macros/src/lib.rs

```diff
@@ -265,6 +266,101 @@ pub fn clone(item: TokenStream) -> TokenStream {
     clone::clone_inner(item)
 }
 
+/// The same as [`gclosure_local!`](crate::gclosure_local) but uses [`Closure::new`] as a
+/// constructor.
```

Comment:
Can you move the docs to this one, which is more generic? And in the other one mention that it's the same but will panic if the closure is called or dropped from a different thread than it was called on?

Reply:
I was going to but couldn't figure out a good way to get the examples/doctests to work without the `local` variant. It seems there would have to be a longer example for `gclosure` with an object subclass that implements `Send` + `Sync`. Would that work?

Reply:
Oh I see. Maybe the examples could stay with the local version (they work the same way anyway), but would be on the non-local version of the macro?

Reply:
Sorry, not sure I understand. Leave the `gclosure_local!` examples on the local version, but also copy the `gclosure_local!` ones to the non-local version?

Reply:
Move the docs you have now on `gclosure_local!` to `gclosure!` and keep the examples in there using `gclosure_local!`, and then put docs on `gclosure_local!` that refer to `gclosure!` for more details.

Reply:
Moved all docs into `closure!` and included some more examples there.